junit-jupiter-api/src/main/kotlin/org/junit/jupiter/api/Assertions.kt

```diff
@@ -134,6 +355,11 @@ inline fun <reified T : Throwable> assertThrows(message: String, executable: ()
  * @see Assertions.assertThrows
  */
 inline fun <reified T : Throwable> assertThrows(noinline message: () -> String, executable: () -> Unit): T {
+    contract {
+        callsInPlace(executable, UNKNOWN)
```

Comment:
Why is this unknown?

Reply:
See these two:
* https://github.com/junit-team/junit5/pull/3259#discussion_r1215525592
* https://github.com/junit-team/junit5/pull/3259#discussion_r1207293456

Reply:
Contracts doesn't work well as being discussed in [this conversation](https://github.com/junit-team/junit5/pull/3259#discussion_r1206850200). 
Perhaps, not specifying contracts for `assertThrows` and `assertTimeoutPreemptively` is a better option. For example, kotlin's [runCatching](https://github.com/JetBrains/kotlin/blob/master/libraries/stdlib/src/kotlin/util/Result.kt#L141) method doesn't specify any contracts.
What do you think?

Reply:
Contracts have been removed for all lambdas that may throw an exception

Reply:
Contracts have been removed for all lambdas that may throw an exception

Reply:
Would it be worth adding a comment into those methods that they intentionally don't have contracts and why?